contracts/test/RFQ.t.sol

```diff
@@ -41,25 +41,50 @@ contract RFQTest is StrategySharedSetup {
 
     address user = vm.addr(userPrivateKey);
     address maker = vm.addr(makerPrivateKey);
-    address feeCollector = address(0x133701);
-    address receiver = address(0x133702);
+    address owner;
+    address feeCollector;
+    address receiver = makeAddr("receiver");
     address[] wallet = [user, maker];
 
     MockERC1271Wallet mockERC1271Wallet;
     MarketMakerProxy marketMakerProxy;
     RFQ rfq;
-    MockWETH weth = new MockWETH("Wrapped ETH", "WETH", 18);
-    IERC20 usdt = new MockERC20("USDT", "USDT", 6);
-    IERC20 dai = new MockERC20("DAI", "DAI", 18);
-    IERC20[] tokens = [weth, usdt, dai];
+    IERC20 weth;
+    IERC20 usdt;
+    IERC20 dai;
+    IERC20[] tokens;
 
     uint256 DEADLINE = block.timestamp + 1;
     RFQLibEIP712.Order DEFAULT_ORDER;
 
     // effectively a "beforeEach" block
     function setUp() public {
         // Setup
-        setUpSystemContracts();
+        if (vm.envBool("deployed")) {
+            weth = IERC20(vm.envAddress("WETH_ADDRESS"));
+            usdt = IERC20(vm.envAddress("USDT_ADDRESS"));
+            dai = IERC20(vm.envAddress("DAI_ADDRESS"));
+
+            allowanceTarget = AllowanceTarget(vm.envAddress("AllowanceTarget_ADDRESS"));
+            spender = Spender(vm.envAddress("Spender_ADDRESS"));
+            userProxy = UserProxy(payable(vm.envAddress("UserProxy_ADDRESS")));
+            permanentStorage = PermanentStorage(vm.envAddress("PermanentStorage_ADDRESS"));
+
+            rfq = RFQ(payable(vm.envAddress("RFQ_ADDRESS")));
+            owner = rfq.owner();
+            feeCollector = rfq.feeCollector();
```

Comment:
Same here. It's little inconvenient to add deployed version support for each tests.

Reply:
I think if these duplication can be avoided and the codes are hidden somewhere where other people writing unit tests do not have to worry about, it's worth the change.

Reply:
I think the `Tokens` extraction is a nice refactoring 👍 In my opinion it would be better to apply the same technique to combine `loadDeployedSystemContracts` and `setUpSystemContracts` to the lower level util contract.

For me `setUpSystemContracts` doesn't imply anything about deploying, it is only saying it will somehow setup system contracts for us to do the testing. So I think embedding `loadDeployedSystemContracts` into it is not a problem.

Reply:
Yeah I thought about it and felt ok with both direction. But RFQ can't have a clean split if these two merged. Checkout 0fbdfd299881d91be419610500f68500e14205ef.